mutt/file.c

```diff
@@ -1335,9 +1333,12 @@ int mutt_file_to_absolute_path(char *path, const char *reference)
   if (!path)
   {
     printf("Error: issue converting path to absolute (%s)", strerror(errno));
+    /* Even fail to get absolute path, FREE() dirpath also */
+    FREE(&dirpath);
     return false;
   }
-
+  /*Beforing return value successfully, first FREE() dirpath */
+  FREE(&dirpath);
   return true;
```

Comment:
These two `FREE()`s stop the leaks, but it would be better if they were closer to the creation.
Make the scope of the variable (and memory allocation) as small as possible.

```c
char *var = alloc_function();
work_function(var);
FREE(&var);
```